### PR TITLE
Issue 646: Add support for providing customized environment variables to controller

### LINKED ISSUE #646: Add support for providing customized environment variables to controller
### Description

Provide support to customize environment variables to controller
### Importance
must-have
### Location

pkg/controller/pravega_controller.go

### Suggestions for an improvement
Provide the required support

### PR DESCRIPTION
**Pull request by anishakj**
Signed-off-by: anisha.kj <anisha.kj@dell.com>

### Change log description

Added support to provide customized env variables to controller pods
### Purpose of the change

Fixes #646

### What the code does

Added new field `ControllerEnvVars` in Pravega spec. Also updated CRD

### How to verify it

Verified that able to provide customized env variables to controller pods

### CODE REVIEW

File: controllers/pravega_controller.go [hunk 1]
```diff
@@ -211,6 +211,9 @@ func makeControllerPodSpec(p *api.PravegaCluster) *corev1.PodSpec {
 	if p.Spec.Pravega.ControllerInitContainers != nil {
 		podSpec.InitContainers = append(podSpec.InitContainers, p.Spec.Pravega.ControllerInitContainers...)
 	}
+	if p.Spec.Pravega.ControllerEnvVars != nil {
+		podSpec.Containers[0].Env = p.Spec.Pravega.ControllerEnvVars
+	}
```
Comment (jkhalack) [inline controllers/pravega_controller.go:216]: Should not we rather append it to the existing array, like it's done for segment store?
https://github.com/pravega/pravega-operator/blob/74333affef845b63c2c1df5c6e138e5888427f45/controllers/pravega_segmentstore.go#L263

Comment (anishakj) [reply controllers/pravega_controller.go:216]: For controller pods, there is no `Env` added to podspec as in segmentstore. That is the reason added like this

Comment (anishakj) [reply controllers/pravega_controller.go:217]: Thanks @jkhalack for pointing this. I have modified it as suggested

**Review by jkhalack — APPROVED** [review]
LGTM

**Review by derekm — APPROVED** [review]

**Pull request merged**